aws_lambda_powertools/event_handler/api_gateway.py

```diff
@@ -1630,15 +1645,14 @@ def _get_openapi_security(
         security: Optional[List[Dict[str, List[str]]]],
         security_schemes: Optional[Dict[str, "SecurityScheme"]],
     ) -> Optional[List[Dict[str, List[str]]]]:
+
         if not security:
             return None
 
-        if not security_schemes:
-            raise ValueError("security_schemes must be provided if security is provided")
-
-        # Check if all keys in security are present in the security_schemes
-        if any(key not in security_schemes for sec in security for key in sec):
-            raise ValueError("Some security schemes not found in security_schemes")
+        if not validate_openapi_security_parameters(security=security, security_schemes=security_schemes):
+            raise SchemaValidationError(
```

Comment:
Hey @heitorlessa! I've updated the exception from `ValueError` to `SchemaValidationError` as it is a schema validation and we should use the appropriate exception name. I don't think that customers are using this, but we can revert in any case. WDYT?

Reply:
great thanks! I'll look into the messaging (what we tell customers in the exception) after lunch